.github/workflows/integration_test_run.yaml

```diff
@@ -171,6 +171,36 @@ jobs:
         )) || inputs.runs-on
       }}
     steps:
+      - uses: actions/checkout@v4.2.2
+      - name: Integration tests variable setting
+        run: |
+          allure_artifact_suffix=$(uuidgen)
+          series=""
+          if [ ! -z ${{ matrix.series }} ]; then
+            series="--series ${{ matrix.series }}"
+            allure_artifact_suffix="$allure_artifact_suffix"-${{ matrix.series }}
+          fi
+          echo "SERIES=$series" >> $GITHUB_ENV
+          module=""
+          if [ ! -z ${{ matrix.modules }} ]; then
+            module="-k ${{ matrix.modules }}"
+            allure_artifact_suffix="$allure_artifact_suffix"-${{ matrix.modules }}
+          fi
+          echo "MODULE=$module" >> $GITHUB_ENV
+          echo "ALLURE_ARTIFACT_SUFFIX=$allure_artifact_suffix" >> $GITHUB_ENV
+      - name: Install tox
+        run: |
+          pip install tox 
+      - name: Collect tests for Allure
```

Comment:
I think maybe we can add a step before this one to check if the Allure as a dependency is declared for this tox test environment and toggle the Allure-related steps on and off accordingly. This way, for projects without Allure installed, it won't run Allure or break the workflow.

For example, something like this (untested):
```
- name: Check Allure
  run: |
    tox -e ${{ inputs.test-tox-env }} --notest --list-dependencies 2>&1 | grep -q allure \
      && echo 'ENABLE_ALLURE=true' >> $GITHUB_ENV \
      || echo 'ENABLE_ALLURE=false' >> $GITHUB_ENV
```
Then use `if: ${{ env.ENABLE_ALLURE == "true" }}` in the following Allure-related steps.

Reply:
addressed